### ISSUE TITLE
[UX] Allow multiple screenshots for modules themes and layouts

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1517]
This is a thought I have been pondering for a while. App stores on both Apple and Android allow you to preview screenshots of your app and what it does. I really like that part of selecting an app. I would like ot see something similar for Backdrop.

We currently allow screenshots for themes, but not (really) for modules or layouts. I would love to be able to handle screenshots consistently across all project types, and deal with them a little better than we do now. I would like to allow for more than one screenshot, so that modules that provide a variety of features could show them all, or themes that have many beautiful pages to display can do so. I propose integration in the following way:

* any project can add a directory `screenshots` that can contain images.
* The GitHub packaging script will remove these screenshots before generating .zip files for download
* The backdropcms.org site will pull in these images, and do the following with them:
    1) The first image (if provided) will be used in the project listing page
    2) All images can be displayed in a carousel or slideshow of some type on the project page
* The in-app project browser can also use these images in the same 2 ways

**Comment by docwilmot** [comment]
Good one. And https://github.com/backdrop-ops/backdropcms.org/issues/100 comes to mind.

**Comment by quicksketch** [comment]
This sounds like a good proposal to me.

**Comment by LeeteqXV** [comment]
Previews are really important (increasingly so as we get more to choose from as we move forward). No time to "test all" just to see which ones may be interesting. This is also a "first impression" for newcomers...

**Comment by opi** [comment]
Nice idea !

> The first image (if provided) will be used in the project listing page

We must discuss a naming convention then. And we all know the [2 hard things in computer science](https://martinfowler.com/bliki/TwoHardThings.html). A maintainer would expect to define the order.

One solution to solve the "1st image" issue, is to use a `screenshot.jpg/png` in project root directory, and then the `screenshots` sub folder.

**Comment by opi** [comment]
*other thoughts*

- allow jpg format, even in [themes](https://api.backdropcms.org/themes) ? discussions can live in another issue if needed.
 - In the theme listing, screenshots are small, and can never be viewed on full screen. we need to solve that! 
- by using a `screenshot.jpg/png` in modules, we'll add an easy first step. The fancy slideshow will come later ;)

> The GitHub packaging script will remove these screenshots before generating .zip files for do

I don't understand why. installed modules will serve image files "locally" instead of b.org or github.com.

**Comment by jenlampton** [comment]
Because full-screen images will be massive in file size, and there's no
need to bloat the software people are using with a tool that would only be
used to select the module/theme in the first place.

**Comment by BWPanda** [comment]
This is much needed I think.

I just watched [this video](https://www.youtube.com/watch?v=IJZr4BTD2Jc) where a guy compares the installation of Drupal 8, Joomla, Wordpress and Backdrop live. One thing he picked up on was how well Wordpress shows the themes you can install with screenshots, whereas Backdrop doesn't have screenshots (and Drupal 8 and Joomla don't even have a built-in theme browser). So we're doing better than Drupal, but not as good as Wordpress yet (in this regard).

**Comment by klonos** [comment]
Just add this ticket to the list of issues for the #1388 meta ;)

**Comment by jenlampton** [comment]
I don't think we have time now for the API changes necessary for adding multiple screenshots to projects. However, since we have the screenshots showing on the themes page (for 1.10) I'm not feeling too terrible about bumping this one out to 1.11.

**Comment by docwilmot** [comment]
Does this need to be a META? Seems like quite a few moving parts. We need changes to core (Project Installer), contrib (although we can update core projects as a best practice example to get things moving), BackdropCMS, and the packaging script. Probably multiple changes each. Lets list this out clearly. @klonos?

**Comment by docwilmot** [comment]
Bumping this, sorry, since @klonos seems to be (temporarily?) back, we need some ~~OCD~~ planning for this or it'll never get done.

**Comment by klonos** [comment]
@docwilmot I never left 😄 ...just time constrains as always. Taking a look now... ish.

**Comment by jenlampton** [comment]
no PR here yet, bumping to 1.12

**Comment by jenlampton** [comment]
no PR here yet, bumping to 1.13

**Comment by jenlampton** [comment]
no PR here yet, bumping to 1.14.

**Comment by jenlampton** [comment]
Removing milestone, adding candidate label.
